benchmarks/dynamo/common.py

```diff
@@ -3015,7 +3016,7 @@ def main(runner, original_dir=None):
         mp.spawn(process_entry, args=(runner, original_dir, args), nprocs=device_count)
     else:
         # single process path just uses the main process
-        args.world_size = 1
+        args.world_size = 1 if args.only else torch.cuda.device_count()
```

Comment:
To clarify, this isnt going to run into the same issue as https://github.com/pytorch/pytorch/issues/107300 (issue with calling `torch.cuda.device_count()` then updating updating `CUDA_VISIBLE_DEVICES`) since the args.only is set to True in cases cases where we update `CUDA_VISIBLE_DEVICES`?

Reply:
I actually don't even need this change since args.world_size gets overwritten to the correct number once we get to the child run

I changed the CUDA_VISIBLE_DEVICES code to only set it for non --multiprocess runs